ITR/data/base_providers.py

```diff
@@ -593,7 +596,8 @@ def _interpolate(self, historic_intensities: pd.DataFrame) -> pd.DataFrame:
         # Interpolate NaNs surrounded by values, and extrapolate NaNs with last known value
         interpolated = historic_intensities.copy()
         for col in interpolated.columns:
-            if interpolated[col].isnull().all():
+            # FIXME: need a unp.isnan that can see into Quantities
+            if unp.isnan(interpolated[col].values.quantity).all():
```

Comment:
why do we change this check from columns to values?
why was original check incorrect?

Reply:
The original check works fine when numbers are floats or quantities of floats.  But quantities of uncertainties don't work.  The .values.quantity extension pulls the values out of the Quantity shell an exposes the ufloats directly to the unp.isnan function.  Now, such an operation discards units, which would be bad if we were using the values of the magnitude.  But all we are checking for here is isnan, and for non-NaN values, we don't care whether it's 10*g CO2 or 10*t CO2.  I.e., in this case the units don't matter.